sdk/identity/identity/src/credentials/managedIdentityCredential/appServiceMsi2017.ts

```diff
@@ -43,6 +44,9 @@ function prepareRequestOptions(
   if (clientId) {
     queryParameters.clientid = clientId;
   }
+  if (resourceId) {
+    queryParameters.mi_res_id = resourceId;
```

Comment:
the AppService MSI api version `2017-09-01` doesn't support this query parameter. We're still discussing if we should validate this and error, or just optimistically send the request. But you're inconsistent in this PR as you error in the Arc case but send the request with the unsupported param here.

Reply:
For now, I will disable these MSIs that don’t support resourceId. Meaning that, if `resourceId` is passed, these MSIs will log a message and they won’t be picked to get a token.